api/src/main/java/org/xwiki/apps/googleapps/GoogleAppsManager.java

```diff
@@ -0,0 +1,242 @@
+     *          or "ok" if successful
+     * @since 3.0
+     */
+    @Unstable
+    String updateUser();
+
+    /**
+     *  Get the list of all documents in the user's associated account.
+     *
+     * @return A list of max 10 documents.
+     * @throws XWikiException if an authorization process failed.
+     * @throws IOException if a communication process to Google services occurred.
+     * @since 3.0
+     */
+    @Unstable
+    List<File> getDocumentList() throws XWikiException, IOException;
+
+    /** Fetches a list of Google Drive document matching a substring query in the filename.
+     *
+     * @param query the expected query (e.g. fullText contains winter ski)
+     * @param nbResults max number of results
+     * @return The list of files at Google Drive.
+     * @throws XWikiException if an XWiki issue occurs
+     * @throws IOException if an error interacting with Google services occurred
+     * @since 3.0
+     */
+    @Unstable
+    List<File> listDriveDocumentsWithTypes(String query, int nbResults) throws XWikiException, IOException;
+
+    /** Fetches a list of Google Drive document matching a given query.
+     *
+     * @param query the expected filename substring
+     * @param nbResults max number of results
+     * @return The list of files at Google Drive.
+     * @throws XWikiException if an XWiki issue occurs
+     * @throws IOException if an error interacting with Google services occurred
+     * @since 3.0
+     */
+    @Unstable
+    FileList listDocuments(String query, int nbResults) throws XWikiException, IOException;
+
+    /**
+     * Fetches the google-drive document's representation and stores it as attachment.
+     * @param page attach to this page
+     * @param name attach using this file name
+     * @param id store object attached to this attachment using this id (for later sync)
+     * @param url fetch from this URL
+     * @return true if successful
+     * @throws XWikiException if an issue occurred in XWiki
+     * @throws IOException if an issue occurred in the communication with teh Google services
+     * @since 3.0
+     */
+    @Unstable
+    boolean retrieveFileFromGoogle(String page, String name, String id, String url) throws XWikiException, IOException;
+
+    /**
+     * Extracts metadata about the Google Drive document corresponding to the named attachment.
+     *
+     * @param pageName The XWiki page where the attachment is
+     * @param fileName The filename of the attachment
+     * @return information about the corresponding Google Drive document
+     * @throws XWikiException if something happened at XWiki side
+     * @since 3.0
+     */
+    @Unstable
+    GoogleAppsManager.GoogleDocMetadata getGoogleDocument(String pageName, String fileName) throws XWikiException;
+
+    /**
+     * Simple pojo for metadata about a google doc.
+     * @since 3.0
+     */
+    @Unstable
+    class GoogleDocMetadata
```

Comment:
Nested classes are usually put at the start of the parent class / interface in order to find them easily.

Reply:
That's one of the CheckStyle comments and honestly I believe that this could just be a top-level class (or do you see anything speaking against it?).
However, I have refrained from such refactorings as they would loose the inner coherence of the proximity of methods which is the only left coherence from the Groovy scripts before we go for a full refactoring where, among others, the complexity of methods and classes will be reduced. I thus suggest to not change this yet.

Reply:
Ok, ignore that. This comment was not appropriate for this and I moved it up.
I'm happy to hear if it is a good idea to move this to a top-level class.

Reply:
Nested classes are usually for internal usage (inside the parent class). If the nested class is meant to be used outside, as it is returned by methods of the parent class, then usually it's best to make it a top level class.

Reply:
This was moved out.
thanks
paul